portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -71,13 +89,24 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+
+    //server side Post check: login required to comment
+    if (!userService.isUserLoggedIn()) {
+      response.sendRedirect("/login_out");
+      return;
+    }
+
     String comment = getParameter(request, COMMENT, "");
     String user = getParameter(request, NAME, "");
+    String userEmail = userService.getCurrentUser().getEmail();
     long timestamp = System.currentTimeMillis();
     
+    
```

Comment:
nit: Please remove one of these empty lines.

Reply:
done